### PR TITLE
LPS-119068 [Page Audit] Update configuration to support the Page Speed Api Key

### PR DESCRIPTION
**Pull request by cgoncas**
**Motivation**

- Only users with View Permission to view the page should see the icon
- If the user has permission to navigate to System Settings, then a bottom should be visible to configure the panel.

**Proposed Solution**

- Add the String field API Key, to the current LayoutReportsConfiguration.
- Add a new property to the context of the React component with the link of the button.

**How to test it**

Integration tests are included in this PR

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:reevaluate:1187401_5514

**Comment by cgoncas** [comment]
ci:reevaluate:1187401_5514

**Comment by cgoncas** [comment]
ci:reevaluate:1187401_5514

**Comment by cgoncas** [comment]
ci:test:relevant

**Review by dgarciasarai — APPROVED** [review]
Hey @cgoncas, thank you so much for changing the JS code as-well. I see nothing to change within this pull request so we are ready to forward it once the CI is green 😄 ✅ 

Screenshot 2021-03-17 at 10 19 12 Screenshot 2021-03-17 at 10 19 32

Thanks!!

**Comment by dgarciasarai** [comment]
One thing that I'm seeing is that the `canonicalURL` for the virtual instance foo.com is starting with '/' and with the `text-truncate-reverse` CSS class is taking the slash and put it as the first letter. Maybe something missing/wrong in the backend side? For localhost is ok. cc: @cgoncas

File: modules/apps/layout/layout-reports-web/test/js/components/EmptyLayoutReports.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ describe('EmptyLayoutReports', () => {
 		const {getByText} = render(
 			<EmptyLayoutReports
 				assetsPath={testProps.assetsPath}
-				showButton={true}
+				configurePageSpeedURL={"validURL"}
```
Comment (dgarciasarai) [inline modules/apps/layout/layout-reports-web/test/js/components/EmptyLayoutReports.js:54]: Curly braces are unnecessary here

**Comment by cgoncas** [comment]
> One thing that I'm seeing is that the `canonicalURL` for the virtual instance foo.com is starting with '/' and with the `text-truncate-reverse` CSS class is taking the slash and put it as the first letter. Maybe something missing/wrong in the backend side? For localhost is ok. cc: @cgoncas

Hey @dgarciasarai !!

Could you put the steps to reproduce this bug?

I have tried with:

Screenshot 2021-03-17 at 14 39 32

**Comment by cgoncas** [comment]
Hey @dgarciasarai !!

I think that we should not use a Button when we need to navigate, so I have used ClayLink. Do you mind to update the CSS to it looks like the Figma desgin?

Thanks!!

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Hey @cgoncas,

> Could you put the steps to reproduce this bug?

I have a virtual instance and I access "http://foo.com:8080/" and I see the image that I attached before. I didn't do anything else.

> I think that we should not use a Button when we need to navigate, so I have used ClayLink. Do you mind to update the CSS to it looks like the Figma design?

Sure! I'm going to take a look and I'll update the pull request.

Thanks!

**Comment by dgarciasarai** [comment]
ci:stop

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Hey @cgoncas, I added the CSS classes necessaries to style the `ClayLink` in this commit https://github.com/liferay-tango/liferay-portal/pull/997/commits/7c046415838075fda1d7a03cbe438ffeaabb1303. Thanks!

**Comment by cgoncas** [comment]
> Hey @cgoncas, I added the necessary CSS classes to style the `ClayLink` in this commit [7c04641](https://github.com/liferay-tango/liferay-portal/commit/7c046415838075fda1d7a03cbe438ffeaabb1303). Thanks!

Thank you very much :)

**Comment by cgoncas** [comment]
Hey @

> Hey @cgoncas,
> 
> > Could you put the steps to reproduce this bug?
> 
> I have a virtual instance and I access "http://foo.com:8080/" and I see the image that I attached before. I didn't do anything else.
> 
> > I think that we should not use a Button when we need to navigate, so I have used ClayLink. Do you mind to update the CSS to it looks like the Figma design?
> 
> Sure! I'm going to take a look and I'll update the pull request.
> 
> Thanks!

Hey @dgarciasarai !!

Do you mind open a bug with this behaviour?

I think the same is happening in the Content Performance Panel.

Thanks!!

**Comment by dgarciasarai** [comment]
Hey @cgoncas, here it is https://issues.liferay.com/browse/LPS-129321. Thanks!

**Comment by cgoncas** [comment]
> Hey @cgoncas, here it is https://issues.liferay.com/browse/LPS-129321. Thanks!

That is awesome @dgarciasarai !!! thank you very much :)

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**